storage/storage-interface/src/async_proof_fetcher.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{proof_fetcher::ProofFetcher, DbReader};
+
+use anyhow::Result;
+use aptos_crypto::{hash::CryptoHash, HashValue};
+use aptos_types::{
+    proof::SparseMerkleProof,
+    state_store::{state_key::StateKey, state_value::StateValue},
+    transaction::Version,
+};
+use crossbeam_channel::{unbounded, Receiver, Sender};
+use num_cpus;
+use once_cell::sync::Lazy;
+use std::{
+    cmp::min,
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+static IO_POOL: Lazy<rayon::ThreadPool> = Lazy::new(|| {
+    rayon::ThreadPoolBuilder::new()
+        // TODO(grao): Do more analysis to tune this magic number.
+        .num_threads(min(256, 4 * num_cpus::get()))
```

Comment:
The theory is this is limited by the concurrency in the storage device? In that case shall we make it 16 or 32 at the minimum?

Reply:
synced offline, let's make it a config item